src/Data/Biapplicative.hs

```diff
@@ -84,6 +88,40 @@ biliftA3 :: Biapplicative w => (a -> b -> c -> d) -> (e -> f -> g -> h) -> w a e
 biliftA3 f g a b c = bimap f g <<$>> a <<*>> b <<*>> c
 {-# INLINE biliftA3 #-}
 
+data Mag a b t where
```

Comment:
A brief comment describing what this is used for might be helpful for readers. AFAICT, this is used to reify the structure of a `traverse` for `traverse2`'s purposes, yes?

Reply:
Yes.

Reply:
I should probably also move it under `traverse2` (or whatever we end up calling it).